### PR TITLE
Support for parralel queries

### LINKED ISSUE #1: Parallel queries from threads
Could you please briefly describe how the parallel queries will work?

### PR DESCRIPTION
**Pull request by akhilman**
Make possible to use ediсt in multithreaded applications.

* [x] Make World::epoch atomic;
* [x] ~~Make sure queries works as expected when the epoch overflows;~~
* [ ] Implement query_mut_unchecked, query_one_mut_unchecked, for_each_mut_unchecked, for_each_tracked_mut_unchecked, get_mut_unchecked unsafe methods for World;
* [ ] Add feature flag for `World` and `Component` `Send+Sync`.
* [ ] Optional: Add a runtime check for aliasing like hecs does by Fetch::borrow and Fetch::release;
* [ ] Optional: Use 32-bit atomic on platforms that do not support 64-bit atomics.

Resolves #1

### CODE REVIEW

File: src/world/mod.rs [hunk 1]
```diff
@@ -193,8 +194,8 @@ impl World {
         let archetype_idx =
             cached_archetype_idx(&mut self.keys, &mut self.ids, &mut self.archetypes, &bundle);
 
-        self.epoch += 1;
-        let idx = self.archetypes[archetype_idx as usize].spawn(entity, bundle, self.epoch);
+        let epoch = self.epoch.fetch_add(1, Ordering::Relaxed).wrapping_add(1);
```
Comment (zakarumych) [inline src/world/mod.rs]: You don't need to perform increments atomically in methods that takes `&mut self`.
Use `AtomicU64::get_mut` to take a mutable reference to the underlying integer and mutate as usual.

**Comment by akhilman** [comment]
Can we just copy the borrow check implementation from hecs and add a link to original repository into the file header?
https://github.com/Ralith/hecs/blob/master/src/borrow.rs

**Comment by zakarumych** [comment]
Sure, why not

**Comment by zakarumych** [comment]
Multithreading support and out-of-the-box scheduler is implemented.
Closing this PR.

**Pull request closed without merging**